### PR TITLE
Add Support for UpDownCounter, ObservableUpDownCounter, ObservableGauge<int64_t>, Histogram<double>

### PR DESCRIPTION
**Pull request by lalitb**
Add support for
 - UpDownCounter (signed integer is converted to double)
 - ObservableUpDownCounter ( signed integer is converted to double)
 - Observable Gauge of signed integer type ( signed integer is converted to double)
 - Histogram of double type (double is converted to uint64_t before transmission)

### CODE REVIEW

File: exporters/geneva/include/opentelemetry/exporters/geneva/metrics/exporter.h [hunk 1]
```diff
@@ -33,6 +33,21 @@ constexpr uint64_t kSecondsToUnixTime =
                   // 1601-01-01T00:00:00Z and UNIX/Linux epoch
                   // (1970-01-01T00:00:00Z)
 
+using ValueType = nostd::variant<int64_t, double>;
+
+enum class MetricsEventType : uint16_t {
```
Comment (ThomsonTan) [inline exporters/geneva/include/opentelemetry/exporters/geneva/metrics/exporter.h:38]: Is there a enum definition for default value like 0？

File: exporters/geneva/src/exporter.cc [hunk 2]
```diff
@@ -78,50 +78,85 @@ opentelemetry::sdk::common::ExportResult Exporter::Export(
...
-                point_data_with_attributes.point_data)) {
+                point_data_with_attributes.point_data)) 
+        {
           auto value = nostd::get<sdk::metrics::SumPointData>(
               point_data_with_attributes.point_data);
-          MetricsEventType event_type = MetricsEventType::ULongMetric;
-          if (nostd::holds_alternative<double>(value.value_)) {
+          ValueType new_value = value.value_;
+
+          MetricsEventType event_type = MetricsEventType::Undefined;
```
Comment (ThomsonTan) [inline exporters/geneva/src/exporter.cc:87]: This assignment is not needed as event_type is always assigned below？

File: exporters/geneva/src/exporter.cc [hunk 3]
```diff
@@ -78,50 +78,85 @@ opentelemetry::sdk::common::ExportResult Exporter::Export(
...
-            LOG_ERROR("Geneva Exporter::Export- Double value is not supported "
-                      "for Histogram");
-            continue;
+          ValueType new_sum = value.sum_;
+          ValueType new_min = value.min_;
+          ValueType new_max = value.max_;
+
+          if (nostd::holds_alternative<double>(value.sum_)) 
+          {
+            // Double is not supported by Geneva, convert it to int64_t
```
Comment (ThomsonTan) [inline exporters/geneva/src/exporter.cc:150]: Will it be supported later? If so, probably add a TODO?

File: exporters/geneva/include/opentelemetry/exporters/geneva/metrics/exporter.h (hunk 1 above)
Comment (lalitb) [reply exporters/geneva/include/opentelemetry/exporters/geneva/metrics/exporter.h:38]: `0` is defined as old_ifx value, but it is not something default 
https://github.com/open-telemetry/opentelemetry-cpp-contrib/blob/dba13fef65749f0fc703076a05896c7f28c44976/exporters/geneva/test/decoder/ifx_metrics_bin.ksy#L221-L223

File: exporters/geneva/src/exporter.cc (hunk 2 above)
Comment (lalitb) [reply exporters/geneva/src/exporter.cc]: Yes, removed the assignment now.

File: exporters/geneva/src/exporter.cc (hunk 3 above)
Comment (lalitb) [reply exporters/geneva/src/exporter.cc]: Yes, we have to modify our code once Geneva start supporting double. Have added TODO for now.

File: exporters/geneva/src/exporter.cc [hunk 4]
```diff
@@ -232,14 +267,18 @@ size_t Exporter::SerializeNonHistogramMetrics(
           .count());
 
   SerializeInt<uint64_t>(buffer_non_histogram_, bufferIndex, windows_ticks);
-  if (event_type == MetricsEventType::ULongMetric) {
+  if (event_type == MetricsEventType::Uint64Metric) {
     SerializeInt<uint64_t>(buffer_non_histogram_, bufferIndex,
                            static_cast<uint64_t>(nostd::get<int64_t>(value)));
-  } else {
+  } else if (event_type == MetricsEventType::DoubleMetric)
+  {
```
Comment (ThomsonTan) [inline exporters/geneva/src/exporter.cc:274]: nit: join this with above line?

File: exporters/geneva/test/common/generate_metrics.h [hunk 5]
```diff
@@ -111,6 +114,96 @@ GenerateSumDataLongMetrics() {
   return data;
 }
 
+// Counter Instrument of long type
+const std::string kUpDownCounterLongInstrumentName =
```
Comment (ThomsonTan) [inline exporters/geneva/test/common/generate_metrics.h:118]: nit: seems these string constants are only used once? If so, put them in in the function argument list directly?

**Review by ThomsonTan — APPROVED** [review]

File: exporters/geneva/src/exporter.cc (hunk 4 above)
Comment (lalitb) [reply exporters/geneva/src/exporter.cc]: thanks fixed. Also fixed clang formatting for the code in general

File: exporters/geneva/test/common/generate_metrics.h (hunk 5 above)
Comment (lalitb) [reply exporters/geneva/test/common/generate_metrics.h:116]: These strings are used in the unit tests for validation.

**Pull request merged**